# Alignment of whole genome data

## Environmental variables

Before we start today, we are going to learn about environmental variables, and create one that will allow us to skip typing out the path to the course folders. 

Variables are a useful way to store data that can be referenced in other commands. For example, we could store the name of the file `super_long_file_name_37456392347283.txt` in the variable `$FILE`. This would allow us to then refer to the file using the variable, instead of typing out the name: 

```{cat, class.source='foldable'}
FILE=super_long_file_name_37456392347283.txt

wc -l $FILE
```

The variable $FILE will only be available in context in which it was created (such as during one terminal session, or one script). However, there is a way to create variables that will apply to all of your terminal sessions. You do this by modifying your `.bashrc` file. 

Your `.bashrc` file is in your home directory, but you might not be able to see it. Unix systems treat file names beginning with `.` as hidden files. To see them in the terminal, you can use `ls -a ~/`. You will see there are a lot of files here! Using either the terminal or the OnDemand file system, add the following lines to the end of your `.bashrc` file. 

```{cat, class.source='foldable'}
export shared=/scratch/alpine/mefr3284/ebio4100
export turnin=/scratch/alpine/mefr3284/Assignments
```

Now we can use the variables `$shared` and `$turnin` to refer to the folders where you get files for the activities and turn in your homework, respectively. This change will not take effect until you start a new terminal session (i.e. log out and log back in). 

You can set these variable names to whatever makes sense to you, but I will use these two names in the rest of the activities for consistency.  

## Explore the sequencing data and reference genome 

**1. Start by making a new folder in your scratch directory and copying today's file there.**

```{cat, class.source='foldable'}
cd /scratch/alpine/usrname
mkdir day_3
cp $shared/day_3/* day_3/
cd day_3/
ls 
```

**2. Start an interactive job. **

```{cat, class.source='foldable'}
sinteractive --partition=amilan -t 4:00:00
```

**3. Look at the data we will be using today. What kinds of file formats are these? What kind of information do they contain? **

```{cat, class.source='foldable'}
less mountain_chickadee_R1.fastq.gz
```

**4. How many files are in the folder? How many individual read sequences does this correspond to? **

These files are gzipped (a specific type of file compression) to make them more compact (this is what the `.gz` on the end of the filename indicates). The command `less` works because it internally knows how to handle those types of files, but not all commands do. One way to deal with this is to use `zcat`. The command `cat` can print out the content of a file; `zcat` does the same, but in a way that can handle zipped files. 

```{cat, class.source='foldable'}
# how many fastq files
zcat mountain_chickadee_R1.fastq.gz | wc -l 
zcat mountain_chickadee_R2.fastq.gz | wc -l 
```

Fastq files have 4 lines for each read, so the number of lines divided by 4 should give you the number of reads. 

**5. What files are present in the reference genome folder?**

```{cat, class.source='foldable'}
ls $shared/reference
```

**6. What are the scaffolds of the chickadee reference genome?** 
```{cat, class.source='foldable'}
grep '>' $shared/reference/moch.dovetail.reference.hap1.scaffolds.renamed.fasta
```

## FASTQC 

The first step after getting new sequencing data (or any type of data) is to do quality control (QC). We can't go through all of this information by hand, so we need to start analyzing this data in other ways. For fastq files, we can check them using FastQC. 

If we wanted to use FastQC on our own computers, we would need to download and install the program. The program is already available on the cluster, but it is stored in something called a module. 

**7. View the available modules and load FastQC.**

```{cat, class.source='foldable'}
module avail
module load fastqc 
```

::: {.caution data=latex=""}
On the Alpine cluster, modules are not available to be loaded on the login nodes. You should already be in an interactive job, but if you cannot view the list of available modules, you should check your location. 
:::

**8. Run FastQC on the first file. **

```{cat, class.source='foldable'}
fastqc mountain_chickadee_R1.fastq.gz 
```

**9. What output files does it make? **

```{cat, class.source='foldable'}
ls
```

Take a look at the html output file (you may need to download it from the OnDemand Portal). How do the results look? 

If you want more information on any of the stats, check out the [documentation here.](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/) This site also has examples of [good](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and [bad](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) data. 

**10. Repeat fastqc with the second set of reads. How do the results compare? **

```{cat, class.source='foldable'}
fastqc mountain_chickadee_R2.fastq.gz 
```

## Trimming 

**11. Use trimmomatic to trim the reads. **

When you load the module, it will give you cluster-specific instructions for running the program.

```{cat, class.source='foldable'}
module load trimmomatic 

java -jar $TRIMMOMATIC PE mountain_chickadee_R1.fastq.gz mountain_chickadee_R2.fastq.gz -trimlog mountain.trimlog -baseout mountain_chickadee_trimmed ILLUMINACLIP:$ADAPTERS/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:20 MINLEN:75
```

Let's break down this command.

`java -jar $TRIMMOMATIC` is our program. Here, `$TRIMMOMATIC` is an environmental variable set by the module when we loaded it. It tells the computer where to find the trimmomatic file. 

`PE` tells trimmomatic that we are going to give it paired end reads. 

`mountain_chickadee_R1.fastq.gz mountain_chickadee_R2.fastq.gz` are our input fastq files that we want it to trim. 

`-trimlog mountain.trimlog` names an file for trimmomatic to use as a log. This will give us information on each read that it processes. 

`-baseout mountain_chickadee_trimmed` will give this prefix to all the output fastqs. 

`ILLUMINACLIP:$ADAPTERS/TruSeq3-PE.fa:2:30:10` tells trimmomatic how to deal with adapter sequences. First, `$ADAPTERS/TruSeq3-PE.fa` gives the location of the adapter file. These are stored on the cluster, and `$ADAPTERS` is another variable created when we loaded the module. The numbers provide thresholds for how closely the read must match the adapter sequence to be trimmed. 

`SLIDINGWINDOW:5:20` defines quality filtering. Trimmomatic will go across the read in windows of 5 basepairs, and trim sites with a quality score below 20. 

`MINLEN:75` tells trimmomatic to throw out any read that is less than 75 basepairs after trimming. 

There are other options we can use to tune our output, which you can read about [here](https://github.com/usadellab/Trimmomatic).

After running this command, what output files have been made? 

**11. Rerun fastqc on one of the new files. Did it improve?** 

```{cat, class.source='foldable'}
fastqc mountain_chickadee_trimmed_1P  
```

::: {.learning data=latex=""}
A useful program called [MultiQC](https://github.com/MultiQC/MultiQC) can help visualize results for multiple steps of bioinformatic quality control. We won't be using it in this course, but you can check it out [here](https://github.com/MultiQC/MultiQC).
:::


## Alignment with BWA 

**12. Load bwa and samtools modules.**

```{cat, class.source='foldable'}
module load bwa
module load samtools 
```

**13. Align the reads with `bwa mem`, and then use samtools view to remove the low quality alignments. **

```{cat, class.source='foldable'}
bwa mem -t 1 -M -R '@RG\tID:moch\tLB:moch\tPL:ILLUMINA\tSM:moch' $shared/reference/moch.dovetail.reference.hap1.scaffolds.renamed.fasta mountain_chickadee_trimmed_1P mountain_chickadee_trimmed_2P | samtools view -q20 -b > mountain_chickadee.filtered.bam
```

We will name this sample `moch`. Names for samples should be as simple as possible, while still being unique. They should not contain spaces.  

Also, the `-b` flag in the `samtools` command tells it that we want our output in the binary BAM format, which will be smaller than the more readable SAM file. 

What output files were made?

**14. If we look at this file, we will notice that the locations of the reads are random. For many downstream analyses, we will want to sort it. We can also do this with SAMtools. The `-T` flag specifies a prefix for temporary files created during sorting. **

```{cat, class.source='foldable'}
samtools view mountain_chickadee.filtered.bam | less 
samtools sort mountain_chickadee.filtered.bam -T tmp_mntchck > mountain_chickadee.filtered.sorted.bam
```

If we look at the file now, we can see that the reads are in order by location. 

::: {.caution data=latex=""}
Remember that we are working with BAM files here, rather than SAM files. BAM files are binary, we we will not be able to read them like text files or view them with less. In this case, we can use samtools to do the conversion for us, using commands like samtools view. 
:::

**15. Here, we did filtering and sorting as separate steps, but they could've been done in a single step. **

You will notice we have also created several versions of this SAM file. Sometimes, you will want to keep these intermediate files for special analyses, but usually we do not need them. You could delete them once you've made your final file, but you could also skip creating these files altogether by using pipes to connect our commands. The pipe (`|`) tells the terminal to pass the output directly into the next command, rather than into a file. 

```{cat, class.source='foldable'}
bwa mem -t 1 -M -R '@RG\tID:moch21\tLB:moch21\tPL:ILLUMINA\tSM:moch21' $shared/reference/moch.dovetail.reference.hap1.scaffolds.renamed.fasta mountain_chickadee_trimmed_1P mountain_chickadee_trimmed_2P | samtools view -q 20 -b | samtools sort -T tmp_mntchck > mountain_chickadee.filtered.sorted.bam
```

Note that the `-b` flag added to the `samtools view` command is what tells `samtools` we want a BAM file output, rather than SAM. 

**16. Finally, most programs will expect BAM files to be indexed. This allows programs to quickly navigate large files. **

```{cat, class.source='foldable'}
samtools index mountain_chickadee.filtered.sorted.bam
```

We can also see that a new file called `mountain_chickadee.filtered.sorted.bam.bai` was created. This is the index and downstream programs will expect it to be in the same folder as the main file. 


## Read Groups 

Earlier we used the -R flag to add a string to our file. This was the read group. Some downstream programs will require read group information. If we want to merge BAM files together, we will also want the reads assigned to groups. If you do not add a group, or if a downstream program requires different group information, you can add or change read groups in a bam file. There are several ways to do this, but below is an example using `samtools`.  

```{cat, class.source='foldable'}

samtools addreplacerg -r "ID:HNJTWDSX3:2" -r "LB:2" -r "PL:ILLUMINA" -r "PU:HNJTWDSX3:2:moch21" -r "SM:moch21" mountain_chickadee.filtered.sorted.bam -o mountain_chickadee.filtered.sorted.newrg.bam 

samtools index mountain_chickadee.filtered.sorted.newrg.bam 

```

## How good was our alignment? 

**17. Use `samtools flagstat` to look at some metrics for our alignment. **

```{cat, class.source='foldable'}

samtools flagstat mountain_chickadee.filtered.sorted.bam > flagstat.txt

cat flagstat.txt

```

What information do we get from this file?

::: {.learning data=latex=""}
As mentioned in lecture, SAM flags store a lot of information about a sequence. If you'd like to learn more about these encodings, check out the [documentation (Page 7)](https://samtools.github.io/hts-specs/SAMv1.pdf) or this [YouTube explainer](https://www.youtube.com/watch?v=qm4NGV79t4o). You can look up the meaning of specific SAM flags [here.](https://broadinstitute.github.io/picard/explain-flags.html)  
::: 

## Assignment 

Create a script that will copy the files `$shared/chickadee_data/MOCH/20H102_S21_L002_R1_001.fastq.gz` and `$shared/chickadee_data/MOCH/20H102_S21_L002_R2_001.fastq.gz` to your scratch directory (you probably want to make a specific folder for this), and then trim, align, and sort the bam file. You can add the read group using the same read group information we used above. Try running the script, which will probably run past the end of class. Copy the script to the folder `$turnin/day_3/` to turn it in. (Add your name to the file name!)

*Hint 1:*
```{cat, class.source='foldable'}
The header of your script should look something like: 
#!/bin/bash

#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --time=6:00:00
#SBATCH --partition=amilan 
#SBATCH --output=bamfiles.out
#SBATCH --qos=normal
```

*Hint 2:*
```{cat, class.source='foldable'}
Do not forget to load the modules you need. 
```

*Hint 3:*
```{cat, class.source='foldable'}
To run a script, use: 
sbatch script_name.sh
```


::: {.learning data=latex=""}
If you would like more information on the programs we tried today, here are links to check out. 
[Trimmomatic Documentation](https://github.com/usadellab/Trimmomatic)

[BWA Documentation](https://bio-bwa.sourceforge.net/bwa.shtml)

[Samtools Documentation](https://www.htslib.org/doc/samtools.html)

[BCFtools Manual](https://samtools.github.io/bcftools/bcftools.html)

::: 