<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Activity 6 Basic Phylogenetics | EBIO4100: Bioinformatics in the Mountains</title>
  <meta name="description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Activity 6 Basic Phylogenetics | EBIO4100: Bioinformatics in the Mountains" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/logo_hagar.JPG" />
  <meta property="og:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="github-repo" content="mfrayer/bioinformatics_in_the_mountains" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Activity 6 Basic Phylogenetics | EBIO4100: Bioinformatics in the Mountains" />
  
  <meta name="twitter:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="twitter:image" content="/images/logo_hagar.JPG" />

<meta name="author" content="Megan Frayer" />


<meta name="date" content="2025-07-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="variant-filtering.html"/>
<link rel="next" href="annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EBIO4100: Bioinformatics in the Mountains</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><i class="fa fa-check"></i>Welcome to EBIO4100: Bioinformatics in the Mountains!</a>
<ul>
<li class="chapter" data-level="0.1" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-use-this-ebook"><i class="fa fa-check"></i><b>0.1</b> How to use this ebook</a></li>
<li class="chapter" data-level="0.2" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#goalslearning-objectives"><i class="fa fa-check"></i><b>0.2</b> Goals/Learning Objectives</a></li>
<li class="chapter" data-level="0.3" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-succeed-in-this-course"><i class="fa fa-check"></i><b>0.3</b> How to succeed in this course:</a></li>
<li class="chapter" data-level="0.4" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#important-links"><i class="fa fa-check"></i><b>0.4</b> Important Links</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>1</b> Computer set up and Introduction to the command line</a>
<ul>
<li class="chapter" data-level="1.1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#get-oriented-to-the-cluster"><i class="fa fa-check"></i><b>1.1</b> Get oriented to the cluster</a></li>
<li class="chapter" data-level="1.2" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#first-command"><i class="fa fa-check"></i><b>1.2</b> First command</a></li>
<li class="chapter" data-level="1.3" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#how-to-navigate-a-file-system"><i class="fa fa-check"></i><b>1.3</b> How to navigate a file system</a></li>
<li class="chapter" data-level="1.4" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#basic-file-manipulation"><i class="fa fa-check"></i><b>1.4</b> Basic file manipulation</a></li>
<li class="chapter" data-level="1.5" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#assignment"><i class="fa fa-check"></i><b>1.5</b> Assignment</a></li>
<li class="chapter" data-level="1.6" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#if-we-have-extra-time"><i class="fa fa-check"></i><b>1.6</b> If we have extra time…</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html"><i class="fa fa-check"></i><b>2</b> FASTA files and visual alignment with MAFFT</a>
<ul>
<li class="chapter" data-level="2.1" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#exploring-a-fasta-file"><i class="fa fa-check"></i><b>2.1</b> Exploring a fasta file</a></li>
<li class="chapter" data-level="2.2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#visual-alignment"><i class="fa fa-check"></i><b>2.2</b> Visual Alignment</a></li>
<li class="chapter" data-level="2.3" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#mafft-multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.3</b> MAFFT Multiple Sequence Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#assignment-1"><i class="fa fa-check"></i><b>2.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html"><i class="fa fa-check"></i><b>3</b> Alignment of whole genome data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#environmental-variables"><i class="fa fa-check"></i><b>3.1</b> Environmental variables</a></li>
<li class="chapter" data-level="3.2" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#explore-the-sequencing-data-and-reference-genome"><i class="fa fa-check"></i><b>3.2</b> Explore the sequencing data and reference genome</a></li>
<li class="chapter" data-level="3.3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#fastqc"><i class="fa fa-check"></i><b>3.3</b> FASTQC</a></li>
<li class="chapter" data-level="3.4" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#trimming"><i class="fa fa-check"></i><b>3.4</b> Trimming</a></li>
<li class="chapter" data-level="3.5" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#alignment-with-bwa"><i class="fa fa-check"></i><b>3.5</b> Alignment with BWA</a></li>
<li class="chapter" data-level="3.6" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#read-groups"><i class="fa fa-check"></i><b>3.6</b> Read Groups</a></li>
<li class="chapter" data-level="3.7" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#how-good-was-our-alignment"><i class="fa fa-check"></i><b>3.7</b> How good was our alignment?</a></li>
<li class="chapter" data-level="3.8" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#assignment-2"><i class="fa fa-check"></i><b>3.8</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>4</b> Variant calling</a>
<ul>
<li class="chapter" data-level="4.1" data-path="variant-calling.html"><a href="variant-calling.html#remove-duplicates-from-the-bam-file"><i class="fa fa-check"></i><b>4.1</b> Remove duplicates from the BAM file</a></li>
<li class="chapter" data-level="4.2" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-gatk"><i class="fa fa-check"></i><b>4.2</b> Call Variants with GATK</a></li>
<li class="chapter" data-level="4.3" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-bcftools"><i class="fa fa-check"></i><b>4.3</b> Call variants with bcftools</a></li>
<li class="chapter" data-level="4.4" data-path="variant-calling.html"><a href="variant-calling.html#are-they-all-the-same"><i class="fa fa-check"></i><b>4.4</b> Are they all the same?</a></li>
<li class="chapter" data-level="4.5" data-path="variant-calling.html"><a href="variant-calling.html#assignment-3"><i class="fa fa-check"></i><b>4.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>5</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-filtering.html"><a href="variant-filtering.html#introduction-to-bcftools"><i class="fa fa-check"></i><b>5.1</b> Introduction to BCFtools</a></li>
<li class="chapter" data-level="5.2" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-quality"><i class="fa fa-check"></i><b>5.2</b> Filter for quality</a></li>
<li class="chapter" data-level="5.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-biallelic-snps"><i class="fa fa-check"></i><b>5.3</b> Filter for biallelic SNPs</a></li>
<li class="chapter" data-level="5.4" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-frequency"><i class="fa fa-check"></i><b>5.4</b> Filter for frequency</a></li>
<li class="chapter" data-level="5.5" data-path="variant-filtering.html"><a href="variant-filtering.html#bcftools-query"><i class="fa fa-check"></i><b>5.5</b> bcftools query</a></li>
<li class="chapter" data-level="5.6" data-path="variant-filtering.html"><a href="variant-filtering.html#assignment-4"><i class="fa fa-check"></i><b>5.6</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html"><i class="fa fa-check"></i><b>6</b> Basic Phylogenetics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#upgma-tree"><i class="fa fa-check"></i><b>6.1</b> UPGMA Tree</a></li>
<li class="chapter" data-level="6.2" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#mafft"><i class="fa fa-check"></i><b>6.2</b> MAFFT</a></li>
<li class="chapter" data-level="6.3" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#iq-tree"><i class="fa fa-check"></i><b>6.3</b> IQ-TREE</a></li>
<li class="chapter" data-level="6.4" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#assignment-5"><i class="fa fa-check"></i><b>6.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="annotation.html"><a href="annotation.html"><i class="fa fa-check"></i><b>7</b> Annotation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="annotation.html"><a href="annotation.html#find-features-in-a-sequence"><i class="fa fa-check"></i><b>7.1</b> Find features in a sequence</a></li>
<li class="chapter" data-level="7.2" data-path="annotation.html"><a href="annotation.html#intro-to-bedtools"><i class="fa fa-check"></i><b>7.2</b> Intro to BEDtools</a></li>
<li class="chapter" data-level="7.3" data-path="annotation.html"><a href="annotation.html#gene-ontology"><i class="fa fa-check"></i><b>7.3</b> Gene Ontology</a></li>
<li class="chapter" data-level="7.4" data-path="annotation.html"><a href="annotation.html#assignment-6"><i class="fa fa-check"></i><b>7.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reproducibility.html"><a href="reproducibility.html"><i class="fa fa-check"></i><b>8</b> Reproducibility</a>
<ul>
<li class="chapter" data-level="8.1" data-path="reproducibility.html"><a href="reproducibility.html#how-do-you-draw-shapes"><i class="fa fa-check"></i><b>8.1</b> How do you draw shapes?</a></li>
<li class="chapter" data-level="8.2" data-path="reproducibility.html"><a href="reproducibility.html#practice-annotating-code"><i class="fa fa-check"></i><b>8.2</b> Practice annotating code</a></li>
<li class="chapter" data-level="8.3" data-path="reproducibility.html"><a href="reproducibility.html#version-control-software"><i class="fa fa-check"></i><b>8.3</b> Version Control Software</a></li>
<li class="chapter" data-level="8.4" data-path="reproducibility.html"><a href="reproducibility.html#explore-github"><i class="fa fa-check"></i><b>8.4</b> Explore github</a></li>
<li class="chapter" data-level="8.5" data-path="reproducibility.html"><a href="reproducibility.html#assignment-7"><i class="fa fa-check"></i><b>8.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html"><i class="fa fa-check"></i><b>9</b> BLAST and Genome Browser</a>
<ul>
<li class="chapter" data-level="9.1" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#ncbi-databases"><i class="fa fa-check"></i><b>9.1</b> NCBI Databases</a></li>
<li class="chapter" data-level="9.2" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#blast"><i class="fa fa-check"></i><b>9.2</b> BLAST</a></li>
<li class="chapter" data-level="9.3" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#genome-browser"><i class="fa fa-check"></i><b>9.3</b> Genome Browser</a></li>
<li class="chapter" data-level="9.4" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#download-data-from-ncbi"><i class="fa fa-check"></i><b>9.4</b> Download Data from NCBI</a></li>
<li class="chapter" data-level="9.5" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#assignment-8"><i class="fa fa-check"></i><b>9.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="rna-seq.html"><a href="rna-seq.html"><i class="fa fa-check"></i><b>10</b> RNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="rna-seq.html"><a href="rna-seq.html#processing-the-data"><i class="fa fa-check"></i><b>10.1</b> Processing the data</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="rna-seq.html"><a href="rna-seq.html#quality-control"><i class="fa fa-check"></i><b>10.1.1</b> Quality Control</a></li>
<li class="chapter" data-level="10.1.2" data-path="rna-seq.html"><a href="rna-seq.html#trimming-1"><i class="fa fa-check"></i><b>10.1.2</b> Trimming</a></li>
<li class="chapter" data-level="10.1.3" data-path="rna-seq.html"><a href="rna-seq.html#alignment"><i class="fa fa-check"></i><b>10.1.3</b> Alignment</a></li>
<li class="chapter" data-level="10.1.4" data-path="rna-seq.html"><a href="rna-seq.html#remove-duplicates"><i class="fa fa-check"></i><b>10.1.4</b> Remove duplicates</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="rna-seq.html"><a href="rna-seq.html#differential-expression"><i class="fa fa-check"></i><b>10.2</b> Differential expression</a></li>
<li class="chapter" data-level="10.3" data-path="rna-seq.html"><a href="rna-seq.html#assignment-9"><i class="fa fa-check"></i><b>10.3</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>11</b> Extras</a>
<ul>
<li class="chapter" data-level="11.1" data-path="extras.html"><a href="extras.html#resources-for-further-learning"><i class="fa fa-check"></i><b>11.1</b> Resources for further learning</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EBIO4100: Bioinformatics in the Mountains</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-phylogenetics" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Activity 6</span> Basic Phylogenetics<a href="basic-phylogenetics.html#basic-phylogenetics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="upgma-tree" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> UPGMA Tree<a href="basic-phylogenetics.html#upgma-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>1. Our first activity today will be to build a UPGMA tree by hand from a small multiple sequence alignment (below). </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>&gt;seqA
CGTTACAGCAGTAGGCGAGTGTACT

&gt;seqB
CGTTACAGCAGTAGGCGGGTGTACT

&gt;seqC
CGATACATCAGTAGGCCAGTGTACT

&gt;seqD
CGATACATCAGTAGGCGAGTGTACT

&gt;seqE
CGTTACAGCCGTAGGGGAGTGTACT

&gt;seqF
CGTTACAGCCGTATGCGAGTGTACT

&gt;seqG
CGTTAGAGCAGTAGTCGACAATACG
</code></pre>
</details>
<div class="learning" data="latex=&quot;&quot;">
<p>Here is a <a href="https://www.youtube.com/watch?v=061ukTACDvY&amp;list=PLjPAsbaa7pgGOrSPNBojU6lFCsJj7pvdJ&amp;index=6">useful walk through</a> of how this algorithm works!</p>
</div>
</div>
<div id="mafft" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> MAFFT<a href="basic-phylogenetics.html#mafft" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Earlier this week we used <a href="https://mafft.cbrc.jp/alignment/server/">MAFFT</a> for multiple sequence alignment, but we can also use it to build trees from those alignments.</p>
<p><strong>2. Put this sequence into MAFFT and create a UPGMA tree. Does this match what we made? </strong></p>
<p>Now let’s try to build a tree using the chickadee data from <code>day_2</code> (also now available in the <code>day_6</code> folder).</p>
<p><strong>3. Enter the chickadee data into MAFFT and create a tree. What does it look like?</strong></p>
<p><strong>4. How is the tree different if you do UPGMA vs Neighbor Joining?</strong></p>
</div>
<div id="iq-tree" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> IQ-TREE<a href="basic-phylogenetics.html#iq-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As we discussed in the morning, UPGMA and Neighbor-Joining algorithms make many simplistic assumptions about how sequences can evolve along a tree. Most trees are built using more complicated models. We will use the IQ-TREE web server to try out one of these approaches.</p>
<p>For this step, we will use the <em>Mimulus</em> data available in the <code>day_6</code> folder. These files are in a format we haven’t seen yet called Phyllip.</p>
<p><strong>5. Make a phylogeny on IQ-TREE.</strong></p>
<p>Go to the <a href="https://www.hiv.lanl.gov/content/sequence/IQTREE/iqtree.html">IQ-TREE webserver</a> and put one of the files into the text box. Create a tree with the default settings.</p>
<p><strong>6. Now change the settings (and write down what you did!).</strong></p>
<p>Did the tree change?</p>
<p><strong>7. From the last tree that you made, get the Newick format tree. (Click “Newick” above the image of your tree.) Send it to me with an explanation of your settings.</strong></p>
<p>You can email it or put it in a file on the cluster, whatever is easiest for you.</p>
<p><strong>8. Try running ultrafast bootstraps. What new information does this give us?</strong></p>
<p><strong>9. Now, create a tree with the second Mimulus file. Use the same settings as one of your other trees so that we can make a direct comparison.</strong></p>
<p>Do the trees differ?</p>
<p><strong>10. If you haven’t yet, try rooting the tree using one of the two forms of midpoint rooting.</strong></p>
<p>How does the tree change?</p>
<p><strong>11. Assuming we get it transferred properly before we get to this step, build a tree from the third Mimulus file. Make sure you use the same settings as one of your previous trees so you can make a direct comparison.</strong></p>
<p>How does this file differ from the previous files? How does the tree differ?</p>
<p><strong>12. Hopefully by this point, we can compare the trees everyone made with different settings.</strong></p>
<p>How different are the trees?</p>
<p><strong>13. Using all of the trees we made, what do we think is the relationship between these samples?</strong></p>
</div>
<div id="assignment-5" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Assignment<a href="basic-phylogenetics.html#assignment-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For today’s assignment, I want you to make a tree for a specific region of the chickadee genome. I have provided a script to start from. <a href="https://www.cell.com/current-biology/fulltext/S0960-9822(24)00398-1#mmc2">Check out this paper to find some gene regions that might be interesting!</a></p>
<p>After you have made the tree, put the Newick format tree into a file on the cluster. Put it in <code>$turnin/day_6</code> with your name in the file name.</p>
<div class="caution" data="latex=&quot;&quot;">
<p>This script has a few issues, some of which are intentional for the sake of time. First, variant calling in a small region like this can produce unusual results and should be done with caution. Second, you would likely want to more carefully define the limits of your gene location using annotations, which we will discuss next week. Third, we are using bams that still have duplicates because there was an issue with the black-capped chickadee “nodup” BAMs.</p>
</div>
<p><em>Hint 1:</em></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
The steps you will want to take include: 
  - calling variants 
  - selecting sites to include 
  - making a fasta format file 
  - putting that fasta into an aligner
  - creating a tree
</code></pre>
</details>
<p><em>Hint 2:</em></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>You can do all of this without writing a script, but you can also modify the script I provided!</code></pre>
</details>
<div class="learning" data="latex=&quot;&quot;">
<p>Here is more information on the programs we used today:</p>
<p><a href="https://iqtree.github.io/">IQ-TREE</a></p>
<p><a href="https://mafft.cbrc.jp/alignment/software/tips0.html">MAFFT</a></p>
<p>This is a link to an online Newick format viewer:
<a href="https://etetoolkit.org/treeview/">ETEToolkit</a></p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="variant-filtering.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/07-phylogeny.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection",
    "scroll_highlight": true
  }
});
});
</script>

</body>

</html>
