<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Activity 2 FASTA files and visual alignment with MAFFT | EBIO4100: Bioinformatics in the Mountains</title>
  <meta name="description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Activity 2 FASTA files and visual alignment with MAFFT | EBIO4100: Bioinformatics in the Mountains" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/logo_hagar.JPG" />
  <meta property="og:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="github-repo" content="mfrayer/bioinformatics_in_the_mountains" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Activity 2 FASTA files and visual alignment with MAFFT | EBIO4100: Bioinformatics in the Mountains" />
  
  <meta name="twitter:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="twitter:image" content="/images/logo_hagar.JPG" />

<meta name="author" content="Megan Frayer" />


<meta name="date" content="2025-07-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="computer-set-up-and-introduction-to-the-command-line.html"/>
<link rel="next" href="alignment-of-whole-genome-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EBIO4100: Bioinformatics in the Mountains</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><i class="fa fa-check"></i>Welcome to EBIO4100: Bioinformatics in the Mountains!</a>
<ul>
<li class="chapter" data-level="0.1" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-use-this-ebook"><i class="fa fa-check"></i><b>0.1</b> How to use this ebook</a></li>
<li class="chapter" data-level="0.2" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#goalslearning-objectives"><i class="fa fa-check"></i><b>0.2</b> Goals/Learning Objectives</a></li>
<li class="chapter" data-level="0.3" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-succeed-in-this-course"><i class="fa fa-check"></i><b>0.3</b> How to succeed in this course:</a></li>
<li class="chapter" data-level="0.4" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#important-links"><i class="fa fa-check"></i><b>0.4</b> Important Links</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>1</b> Computer set up and Introduction to the command line</a>
<ul>
<li class="chapter" data-level="1.1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#get-oriented-to-the-cluster"><i class="fa fa-check"></i><b>1.1</b> Get oriented to the cluster</a></li>
<li class="chapter" data-level="1.2" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#first-command"><i class="fa fa-check"></i><b>1.2</b> First command</a></li>
<li class="chapter" data-level="1.3" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#how-to-navigate-a-file-system"><i class="fa fa-check"></i><b>1.3</b> How to navigate a file system</a></li>
<li class="chapter" data-level="1.4" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#basic-file-manipulation"><i class="fa fa-check"></i><b>1.4</b> Basic file manipulation</a></li>
<li class="chapter" data-level="1.5" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#assignment"><i class="fa fa-check"></i><b>1.5</b> Assignment</a></li>
<li class="chapter" data-level="1.6" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#if-we-have-extra-time"><i class="fa fa-check"></i><b>1.6</b> If we have extra time…</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html"><i class="fa fa-check"></i><b>2</b> FASTA files and visual alignment with MAFFT</a>
<ul>
<li class="chapter" data-level="2.1" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#exploring-a-fasta-file"><i class="fa fa-check"></i><b>2.1</b> Exploring a fasta file</a></li>
<li class="chapter" data-level="2.2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#visual-alignment"><i class="fa fa-check"></i><b>2.2</b> Visual Alignment</a></li>
<li class="chapter" data-level="2.3" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#mafft-multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.3</b> MAFFT Multiple Sequence Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#assignment-1"><i class="fa fa-check"></i><b>2.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html"><i class="fa fa-check"></i><b>3</b> Alignment of whole genome data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#environmental-variables"><i class="fa fa-check"></i><b>3.1</b> Environmental variables</a></li>
<li class="chapter" data-level="3.2" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#explore-the-sequencing-data-and-reference-genome"><i class="fa fa-check"></i><b>3.2</b> Explore the sequencing data and reference genome</a></li>
<li class="chapter" data-level="3.3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#fastqc"><i class="fa fa-check"></i><b>3.3</b> FASTQC</a></li>
<li class="chapter" data-level="3.4" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#trimming"><i class="fa fa-check"></i><b>3.4</b> Trimming</a></li>
<li class="chapter" data-level="3.5" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#alignment-with-bwa"><i class="fa fa-check"></i><b>3.5</b> Alignment with BWA</a></li>
<li class="chapter" data-level="3.6" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#read-groups"><i class="fa fa-check"></i><b>3.6</b> Read Groups</a></li>
<li class="chapter" data-level="3.7" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#how-good-was-our-alignment"><i class="fa fa-check"></i><b>3.7</b> How good was our alignment?</a></li>
<li class="chapter" data-level="3.8" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#assignment-2"><i class="fa fa-check"></i><b>3.8</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>4</b> Variant calling</a>
<ul>
<li class="chapter" data-level="4.1" data-path="variant-calling.html"><a href="variant-calling.html#remove-duplicates-from-the-bam-file"><i class="fa fa-check"></i><b>4.1</b> Remove duplicates from the BAM file</a></li>
<li class="chapter" data-level="4.2" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-gatk"><i class="fa fa-check"></i><b>4.2</b> Call Variants with GATK</a></li>
<li class="chapter" data-level="4.3" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-bcftools"><i class="fa fa-check"></i><b>4.3</b> Call variants with bcftools</a></li>
<li class="chapter" data-level="4.4" data-path="variant-calling.html"><a href="variant-calling.html#are-they-all-the-same"><i class="fa fa-check"></i><b>4.4</b> Are they all the same?</a></li>
<li class="chapter" data-level="4.5" data-path="variant-calling.html"><a href="variant-calling.html#assignment-3"><i class="fa fa-check"></i><b>4.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>5</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-filtering.html"><a href="variant-filtering.html#introduction-to-bcftools"><i class="fa fa-check"></i><b>5.1</b> Introduction to BCFtools</a></li>
<li class="chapter" data-level="5.2" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-quality"><i class="fa fa-check"></i><b>5.2</b> Filter for quality</a></li>
<li class="chapter" data-level="5.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-biallelic-snps"><i class="fa fa-check"></i><b>5.3</b> Filter for biallelic SNPs</a></li>
<li class="chapter" data-level="5.4" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-frequency"><i class="fa fa-check"></i><b>5.4</b> Filter for frequency</a></li>
<li class="chapter" data-level="5.5" data-path="variant-filtering.html"><a href="variant-filtering.html#bcftools-query"><i class="fa fa-check"></i><b>5.5</b> bcftools query</a></li>
<li class="chapter" data-level="5.6" data-path="variant-filtering.html"><a href="variant-filtering.html#assignment-4"><i class="fa fa-check"></i><b>5.6</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html"><i class="fa fa-check"></i><b>6</b> Basic Phylogenetics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#upgma-tree"><i class="fa fa-check"></i><b>6.1</b> UPGMA Tree</a></li>
<li class="chapter" data-level="6.2" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#mafft"><i class="fa fa-check"></i><b>6.2</b> MAFFT</a></li>
<li class="chapter" data-level="6.3" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#iq-tree"><i class="fa fa-check"></i><b>6.3</b> IQ-TREE</a></li>
<li class="chapter" data-level="6.4" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#assignment-5"><i class="fa fa-check"></i><b>6.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="annotation.html"><a href="annotation.html"><i class="fa fa-check"></i><b>7</b> Annotation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="annotation.html"><a href="annotation.html#find-features-in-a-sequence"><i class="fa fa-check"></i><b>7.1</b> Find features in a sequence</a></li>
<li class="chapter" data-level="7.2" data-path="annotation.html"><a href="annotation.html#intro-to-bedtools"><i class="fa fa-check"></i><b>7.2</b> Intro to BEDtools</a></li>
<li class="chapter" data-level="7.3" data-path="annotation.html"><a href="annotation.html#gene-ontology"><i class="fa fa-check"></i><b>7.3</b> Gene Ontology</a></li>
<li class="chapter" data-level="7.4" data-path="annotation.html"><a href="annotation.html#assignment-6"><i class="fa fa-check"></i><b>7.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reproducibility.html"><a href="reproducibility.html"><i class="fa fa-check"></i><b>8</b> Reproducibility</a>
<ul>
<li class="chapter" data-level="8.1" data-path="reproducibility.html"><a href="reproducibility.html#how-do-you-draw-shapes"><i class="fa fa-check"></i><b>8.1</b> How do you draw shapes?</a></li>
<li class="chapter" data-level="8.2" data-path="reproducibility.html"><a href="reproducibility.html#practice-annotating-code"><i class="fa fa-check"></i><b>8.2</b> Practice annotating code</a></li>
<li class="chapter" data-level="8.3" data-path="reproducibility.html"><a href="reproducibility.html#version-control-software"><i class="fa fa-check"></i><b>8.3</b> Version Control Software</a></li>
<li class="chapter" data-level="8.4" data-path="reproducibility.html"><a href="reproducibility.html#explore-github"><i class="fa fa-check"></i><b>8.4</b> Explore github</a></li>
<li class="chapter" data-level="8.5" data-path="reproducibility.html"><a href="reproducibility.html#assignment-7"><i class="fa fa-check"></i><b>8.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html"><i class="fa fa-check"></i><b>9</b> BLAST and Genome Browser</a>
<ul>
<li class="chapter" data-level="9.1" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#ncbi-databases"><i class="fa fa-check"></i><b>9.1</b> NCBI Databases</a></li>
<li class="chapter" data-level="9.2" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#blast"><i class="fa fa-check"></i><b>9.2</b> BLAST</a></li>
<li class="chapter" data-level="9.3" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#genome-browser"><i class="fa fa-check"></i><b>9.3</b> Genome Browser</a></li>
<li class="chapter" data-level="9.4" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#download-data-from-ncbi"><i class="fa fa-check"></i><b>9.4</b> Download Data from NCBI</a></li>
<li class="chapter" data-level="9.5" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#assignment-8"><i class="fa fa-check"></i><b>9.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EBIO4100: Bioinformatics in the Mountains</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="fasta-files-and-visual-alignment-with-mafft" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Activity 2</span> FASTA files and visual alignment with MAFFT<a href="fasta-files-and-visual-alignment-with-mafft.html#fasta-files-and-visual-alignment-with-mafft" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="exploring-a-fasta-file" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Exploring a fasta file<a href="fasta-files-and-visual-alignment-with-mafft.html#exploring-a-fasta-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>1. Start by making a new folder in your scratch directory and copying today’s file there.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>cd /scratch/alpine/usrname
mkdir day_2
cp /scratch/alpine/mefr3284/ebio4100/day_2/* day_2/
ls </code></pre>
</details>
<p><strong>2. Look at the file. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>less chickadee.fasta </code></pre>
</details>
<p>How are the headers marked? What information is included there?</p>
<p><strong>3. How many samples are included in this file? </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>grep &quot;&gt;&quot; chickadee.fasta | wc -l </code></pre>
</details>
<p><strong>4. We can search for patterns within the data using grep. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>grep --color=always &#39;TAAACAA&#39; chickadee.fasta | head -10
grep --color=always &#39;TA[TA]TA&#39; chickadee.fasta | head -10</code></pre>
</details>
<p><strong>5. How many times does the pattern TAAACAA appear in these sequences? </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>grep -o &#39;TAAACAA&#39; chickadee.fasta | wc -l</code></pre>
</details>
<p><strong>6. How many potential start codons are there? </strong></p>
<p>Hint: The “start codon” initiates translation of an amino acid chain. The RNA codon is <code>AUG</code>.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>grep -o &#39;ATG&#39; chickadee.fasta | wc -l</code></pre>
</details>
<div class="learning" data="latex=&quot;&quot;">
<p>The command <code>grep</code> is really powerful! If you’re interested in trying more grep commands, <a href="https://bioinformaticsworkbook.org/Appendix/Unix/unix-basics-3grep.html#gsc.tab=0">here’s a useful list of things you can do with grep.</a></p>
</div>
<p><strong>7. How long are each of these sequences?</strong></p>
<p>This sounds like an easy question, but it is harder than it looks! We need to take into account:</p>
<ol style="list-style-type: decimal">
<li>Each read is split across multiple lines.</li>
<li>Each individual is represented by both an identifier line and a sequence line.</li>
<li>We want to count only the length of the sequence lines, but associate them with the sample lines.</li>
</ol>
<p>Using what we have learned so far, this would be challenging. But I think it is useful to think about. The structure of the fasta file (and the possible variations on it) make things challenging, and this problem gets bigger as the file formats get more complicated. That is why in general, it is a great idea to use existing programs to process data, particularly ones that are well documented and have been thoroughly tested.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>Because I do not want to leave the question unanswered, here is a command that will work. It uses awk, which is a really useful linux program for processing.  
  
awk &#39;/^&gt;/{if (l!=&quot;&quot;) print l; print; l=0; next}{l+=length($0)}END{print l}&#39; chickadee.fasta  </code></pre>
</details>
</div>
<div id="visual-alignment" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Visual Alignment<a href="fasta-files-and-visual-alignment-with-mafft.html#visual-alignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>8. In pairs, try to align these two sequences by hand. </strong></p>
<p>These are protein sequences, so each letter represents an amino acid.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
&gt;seq1
HEAGAWGHEE

&gt;seq2
PAWHEAE
</code></pre>
</details>
<p>How many ways can you align these? What if you allow gaps?</p>
</div>
<div id="mafft-multiple-sequence-alignment" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> MAFFT Multiple Sequence Alignment<a href="fasta-files-and-visual-alignment-with-mafft.html#mafft-multiple-sequence-alignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Multiple sequence aligners do as the name suggests– align multiple sequences at once. There are a variety of algorithms implemented in a variety of programs that perform multiple alignment. Today, we will be using MAFFT (Multiple Alignment Fast Fourier Transform), which is one of the most popular programs and can implement multiple algorithms.</p>
<p>Follow this link to the <a href="https://mafft.cbrc.jp/alignment/server/index.html">online implementation of MAFFT</a>.</p>
<div class="learning" data="latex=&quot;&quot;">
<p>We aren’t going to discuss the inner workings of multiple sequence alignment, but here is a video and two papers you can check out if you are interested.</p>
<p><a href="https://www.youtube.com/watch?v=iOZvZJf3BzU">Lecture walking through the process</a></p>
<p><a href="https://www.sciencedirect.com/science/article/pii/S0959440X06000704">An older but comprehensive comparison</a> <span class="citation">(<a href="#ref-edgar_multiple_2006">Edgar and Batzoglou 2006</a>)</span></p>
<p><a href="https://doi.org/10.1093/bib/bbx108">The most recent MAFFT paper</a> <span class="citation">(<a href="#ref-katoh_mafft_2019">Katoh, Rozewicki, and Yamada 2019</a>)</span></p>
</div>
<p><strong>9. Copy and paste the sequences above into the text box and submit it. You should be able to use all of the default settings. </strong></p>
<p>Is this alignment the same one you came up with?</p>
<p><strong>10. Now try running our chickadee dataset.</strong></p>
<p>You will need to download it from the cluster– I recommend using the OnDemand Portal.</p>
<p>Look at the resulting alignment in the View tab. What has been added to the input data?</p>
<p>Do you see any patterns in the data? (It’s okay if you don’t– this is a difficult way to look at data!)</p>
<p>What other information can you get from MAFFT?</p>
</div>
<div id="assignment-1" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Assignment<a href="fasta-files-and-visual-alignment-with-mafft.html#assignment-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Download the new fasta file from MAFFT, and then upload it to the cluster using the OnDemand portal. Copy it to <code>/scratch/alpine/mefr3284/Assignments/day_2/</code>, adding your name to the file name.</p>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-edgar_multiple_2006" class="csl-entry">
Edgar, Robert C., and Serafim Batzoglou. 2006. <span>“Multiple Sequence Alignment.”</span> <em>Current Opinion in Structural Biology</em> 16 (3): 368–73. https://doi.org/<a href="https://doi.org/10.1016/j.sbi.2006.04.004">https://doi.org/10.1016/j.sbi.2006.04.004</a>.
</div>
<div id="ref-katoh_mafft_2019" class="csl-entry">
Katoh, Kazutaka, John Rozewicki, and Kazunori D Yamada. 2019. <span>“<span>MAFFT</span> Online Service: Multiple Sequence Alignment, Interactive Sequence Choice and Visualization.”</span> <em>Briefings in Bioinformatics</em> 20 (4): 1160–66. <a href="https://doi.org/10.1093/bib/bbx108">https://doi.org/10.1093/bib/bbx108</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="computer-set-up-and-introduction-to-the-command-line.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="alignment-of-whole-genome-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-fasta.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection",
    "scroll_highlight": true
  }
});
});
</script>

</body>

</html>
