<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Activity 8 Reproducibility | EBIO4100: Bioinformatics in the Mountains</title>
  <meta name="description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Activity 8 Reproducibility | EBIO4100: Bioinformatics in the Mountains" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/logo_hagar.JPG" />
  <meta property="og:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="github-repo" content="mfrayer/bioinformatics_in_the_mountains" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Activity 8 Reproducibility | EBIO4100: Bioinformatics in the Mountains" />
  
  <meta name="twitter:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="twitter:image" content="/images/logo_hagar.JPG" />

<meta name="author" content="Megan Frayer" />


<meta name="date" content="2025-07-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="annotation.html"/>
<link rel="next" href="blast-and-genome-browser.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EBIO4100: Bioinformatics in the Mountains</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><i class="fa fa-check"></i>Welcome to EBIO4100: Bioinformatics in the Mountains!</a>
<ul>
<li class="chapter" data-level="0.1" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-use-this-ebook"><i class="fa fa-check"></i><b>0.1</b> How to use this ebook</a></li>
<li class="chapter" data-level="0.2" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#goalslearning-objectives"><i class="fa fa-check"></i><b>0.2</b> Goals/Learning Objectives</a></li>
<li class="chapter" data-level="0.3" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-succeed-in-this-course"><i class="fa fa-check"></i><b>0.3</b> How to succeed in this course:</a></li>
<li class="chapter" data-level="0.4" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#important-links"><i class="fa fa-check"></i><b>0.4</b> Important Links</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>1</b> Computer set up and Introduction to the command line</a>
<ul>
<li class="chapter" data-level="1.1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#get-oriented-to-the-cluster"><i class="fa fa-check"></i><b>1.1</b> Get oriented to the cluster</a></li>
<li class="chapter" data-level="1.2" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#first-command"><i class="fa fa-check"></i><b>1.2</b> First command</a></li>
<li class="chapter" data-level="1.3" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#how-to-navigate-a-file-system"><i class="fa fa-check"></i><b>1.3</b> How to navigate a file system</a></li>
<li class="chapter" data-level="1.4" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#basic-file-manipulation"><i class="fa fa-check"></i><b>1.4</b> Basic file manipulation</a></li>
<li class="chapter" data-level="1.5" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#assignment"><i class="fa fa-check"></i><b>1.5</b> Assignment</a></li>
<li class="chapter" data-level="1.6" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#if-we-have-extra-time"><i class="fa fa-check"></i><b>1.6</b> If we have extra time…</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html"><i class="fa fa-check"></i><b>2</b> FASTA files and visual alignment with MAFFT</a>
<ul>
<li class="chapter" data-level="2.1" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#exploring-a-fasta-file"><i class="fa fa-check"></i><b>2.1</b> Exploring a fasta file</a></li>
<li class="chapter" data-level="2.2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#visual-alignment"><i class="fa fa-check"></i><b>2.2</b> Visual Alignment</a></li>
<li class="chapter" data-level="2.3" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#mafft-multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.3</b> MAFFT Multiple Sequence Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#assignment-1"><i class="fa fa-check"></i><b>2.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html"><i class="fa fa-check"></i><b>3</b> Alignment of whole genome data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#environmental-variables"><i class="fa fa-check"></i><b>3.1</b> Environmental variables</a></li>
<li class="chapter" data-level="3.2" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#explore-the-sequencing-data-and-reference-genome"><i class="fa fa-check"></i><b>3.2</b> Explore the sequencing data and reference genome</a></li>
<li class="chapter" data-level="3.3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#fastqc"><i class="fa fa-check"></i><b>3.3</b> FASTQC</a></li>
<li class="chapter" data-level="3.4" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#trimming"><i class="fa fa-check"></i><b>3.4</b> Trimming</a></li>
<li class="chapter" data-level="3.5" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#alignment-with-bwa"><i class="fa fa-check"></i><b>3.5</b> Alignment with BWA</a></li>
<li class="chapter" data-level="3.6" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#read-groups"><i class="fa fa-check"></i><b>3.6</b> Read Groups</a></li>
<li class="chapter" data-level="3.7" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#how-good-was-our-alignment"><i class="fa fa-check"></i><b>3.7</b> How good was our alignment?</a></li>
<li class="chapter" data-level="3.8" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#assignment-2"><i class="fa fa-check"></i><b>3.8</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>4</b> Variant calling</a>
<ul>
<li class="chapter" data-level="4.1" data-path="variant-calling.html"><a href="variant-calling.html#remove-duplicates-from-the-bam-file"><i class="fa fa-check"></i><b>4.1</b> Remove duplicates from the BAM file</a></li>
<li class="chapter" data-level="4.2" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-gatk"><i class="fa fa-check"></i><b>4.2</b> Call Variants with GATK</a></li>
<li class="chapter" data-level="4.3" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-bcftools"><i class="fa fa-check"></i><b>4.3</b> Call variants with bcftools</a></li>
<li class="chapter" data-level="4.4" data-path="variant-calling.html"><a href="variant-calling.html#are-they-all-the-same"><i class="fa fa-check"></i><b>4.4</b> Are they all the same?</a></li>
<li class="chapter" data-level="4.5" data-path="variant-calling.html"><a href="variant-calling.html#assignment-3"><i class="fa fa-check"></i><b>4.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>5</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-filtering.html"><a href="variant-filtering.html#introduction-to-bcftools"><i class="fa fa-check"></i><b>5.1</b> Introduction to BCFtools</a></li>
<li class="chapter" data-level="5.2" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-quality"><i class="fa fa-check"></i><b>5.2</b> Filter for quality</a></li>
<li class="chapter" data-level="5.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-biallelic-snps"><i class="fa fa-check"></i><b>5.3</b> Filter for biallelic SNPs</a></li>
<li class="chapter" data-level="5.4" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-frequency"><i class="fa fa-check"></i><b>5.4</b> Filter for frequency</a></li>
<li class="chapter" data-level="5.5" data-path="variant-filtering.html"><a href="variant-filtering.html#bcftools-query"><i class="fa fa-check"></i><b>5.5</b> bcftools query</a></li>
<li class="chapter" data-level="5.6" data-path="variant-filtering.html"><a href="variant-filtering.html#assignment-4"><i class="fa fa-check"></i><b>5.6</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html"><i class="fa fa-check"></i><b>6</b> Basic Phylogenetics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#upgma-tree"><i class="fa fa-check"></i><b>6.1</b> UPGMA Tree</a></li>
<li class="chapter" data-level="6.2" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#mafft"><i class="fa fa-check"></i><b>6.2</b> MAFFT</a></li>
<li class="chapter" data-level="6.3" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#iq-tree"><i class="fa fa-check"></i><b>6.3</b> IQ-TREE</a></li>
<li class="chapter" data-level="6.4" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#assignment-5"><i class="fa fa-check"></i><b>6.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="annotation.html"><a href="annotation.html"><i class="fa fa-check"></i><b>7</b> Annotation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="annotation.html"><a href="annotation.html#find-features-in-a-sequence"><i class="fa fa-check"></i><b>7.1</b> Find features in a sequence</a></li>
<li class="chapter" data-level="7.2" data-path="annotation.html"><a href="annotation.html#intro-to-bedtools"><i class="fa fa-check"></i><b>7.2</b> Intro to BEDtools</a></li>
<li class="chapter" data-level="7.3" data-path="annotation.html"><a href="annotation.html#gene-ontology"><i class="fa fa-check"></i><b>7.3</b> Gene Ontology</a></li>
<li class="chapter" data-level="7.4" data-path="annotation.html"><a href="annotation.html#assignment-6"><i class="fa fa-check"></i><b>7.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reproducibility.html"><a href="reproducibility.html"><i class="fa fa-check"></i><b>8</b> Reproducibility</a>
<ul>
<li class="chapter" data-level="8.1" data-path="reproducibility.html"><a href="reproducibility.html#how-do-you-draw-shapes"><i class="fa fa-check"></i><b>8.1</b> How do you draw shapes?</a></li>
<li class="chapter" data-level="8.2" data-path="reproducibility.html"><a href="reproducibility.html#practice-annotating-code"><i class="fa fa-check"></i><b>8.2</b> Practice annotating code</a></li>
<li class="chapter" data-level="8.3" data-path="reproducibility.html"><a href="reproducibility.html#version-control-software"><i class="fa fa-check"></i><b>8.3</b> Version Control Software</a></li>
<li class="chapter" data-level="8.4" data-path="reproducibility.html"><a href="reproducibility.html#explore-github"><i class="fa fa-check"></i><b>8.4</b> Explore github</a></li>
<li class="chapter" data-level="8.5" data-path="reproducibility.html"><a href="reproducibility.html#assignment-7"><i class="fa fa-check"></i><b>8.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html"><i class="fa fa-check"></i><b>9</b> BLAST and Genome Browser</a>
<ul>
<li class="chapter" data-level="9.1" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#ncbi-databases"><i class="fa fa-check"></i><b>9.1</b> NCBI Databases</a></li>
<li class="chapter" data-level="9.2" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#blast"><i class="fa fa-check"></i><b>9.2</b> BLAST</a></li>
<li class="chapter" data-level="9.3" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#genome-browser"><i class="fa fa-check"></i><b>9.3</b> Genome Browser</a></li>
<li class="chapter" data-level="9.4" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#download-data-from-ncbi"><i class="fa fa-check"></i><b>9.4</b> Download Data from NCBI</a></li>
<li class="chapter" data-level="9.5" data-path="blast-and-genome-browser.html"><a href="blast-and-genome-browser.html#assignment-8"><i class="fa fa-check"></i><b>9.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="rna-seq.html"><a href="rna-seq.html"><i class="fa fa-check"></i><b>10</b> RNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="rna-seq.html"><a href="rna-seq.html#processing-the-data"><i class="fa fa-check"></i><b>10.1</b> Processing the data</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="rna-seq.html"><a href="rna-seq.html#quality-control"><i class="fa fa-check"></i><b>10.1.1</b> Quality Control</a></li>
<li class="chapter" data-level="10.1.2" data-path="rna-seq.html"><a href="rna-seq.html#trimming-1"><i class="fa fa-check"></i><b>10.1.2</b> Trimming</a></li>
<li class="chapter" data-level="10.1.3" data-path="rna-seq.html"><a href="rna-seq.html#alignment"><i class="fa fa-check"></i><b>10.1.3</b> Alignment</a></li>
<li class="chapter" data-level="10.1.4" data-path="rna-seq.html"><a href="rna-seq.html#remove-duplicates"><i class="fa fa-check"></i><b>10.1.4</b> Remove duplicates</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="rna-seq.html"><a href="rna-seq.html#differential-expression"><i class="fa fa-check"></i><b>10.2</b> Differential expression</a></li>
<li class="chapter" data-level="10.3" data-path="rna-seq.html"><a href="rna-seq.html#assignment-9"><i class="fa fa-check"></i><b>10.3</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>11</b> Extras</a>
<ul>
<li class="chapter" data-level="11.1" data-path="extras.html"><a href="extras.html#resources-for-further-learning"><i class="fa fa-check"></i><b>11.1</b> Resources for further learning</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EBIO4100: Bioinformatics in the Mountains</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="reproducibility" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Activity 8</span> Reproducibility<a href="reproducibility.html#reproducibility" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="how-do-you-draw-shapes" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> How do you draw shapes?<a href="reproducibility.html#how-do-you-draw-shapes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will do this activity together.</p>
<p><strong>1. Make a folder for today. Then, start an interactive job.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>cd /scratch/alpine/usrname
mkdir day_8
cd day_8

sinteractive --partition=amilan -t 4:00:00</code></pre>
</details>
</div>
<div id="practice-annotating-code" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Practice annotating code<a href="reproducibility.html#practice-annotating-code" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Below is a script for processing read data like we did last week.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
#!/bin/bash

#SBATCH --nodes=1 --ntasks=1
#SBATCH --time=24:00:00
#SBATCH --partition=amilan 
#SBATCH --output=vcfs.out

module purge
module load picard 
module load samtools
module load bcftools 

for birds in /projects/mefr3284/bamfiles/*.filtered.sorted.bam ; do
        out=$(basename ${birds%%.*})
        picard MarkDuplicates -INPUT $birds -OUTPUT ../bamfiles/$out.filtered.sorted.nodup.bam -METRICS_FILE $out.duplicate.metrics -REMOVE_DUPLICATES true -ASSUME_SORTED true -TMP_DIR tmp_$out -VALIDATION_STRINGENCY LENIENT 
        samtools index ../bamfiles/$out.filtered.sorted.nodup.bam
        bcftools mpileup -f /scratch/alpine/mefr3284/ebio4100/reference/moch.dovetail.reference.hap1.scaffolds.renamed.fasta ../bamfiles/$out.filtered.sorted.nodup.bam -a DP,AD -Ou | bcftools call -mv -Oz -f GQ -o /scratch/alpine/mefr3284/vcf/$out.vcf.gz
        tabix -fp vcf /scratch/alpine/mefr3284/vcf/$out.vcf.gz
done
</code></pre>
</details>
<p><strong>2. Copy and paste this script into a file, and practice annotating it.</strong></p>
<p>Suggestions: Include notes on where the data is coming from, document the version of the module that you used, document the steps of the script.</p>
<p>You can find the version of the modules by looking in <code>module avail</code> or by loading the module and running something like <code>bcftools --version</code>.</p>
<p><strong>3. When you think you are done, give your script to a classmate. Read your classmates annotations and make any suggestions where you think something is unclear or missing. Could you easily figure out what this code did if you hadn’t seen it before?</strong></p>
<div class="learning" data="latex=&quot;&quot;">
<p>A great place to start for more information on reproducibility is Karl Broman’s <a href="https://kbroman.org/steps2rr/">Steps Toward Reproducible Research</a> tutorial and the <a href="https://kbroman.org/steps2rr/pages/resources.html">associated resources page</a>!</p>
</div>
</div>
<div id="version-control-software" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Version Control Software<a href="reproducibility.html#version-control-software" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>While there are many types of version control software, we will learn a bit about Git today because it is the most widely used among academic bioinformaticians.</p>
<p><strong>4. Load the git module.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>module load git</code></pre>
</details>
<p><strong>5. Turn your day 7 folder into a git repository. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>git init </code></pre>
</details>
<p>You will get a warning about the branch name. Feel free to rename it to main if you choose.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>git branch -m main</code></pre>
</details>
<p><strong>6. What does your repository look like? </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>git status</code></pre>
</details>
<p><strong>7. Make a file that you can track. </strong></p>
<p>We often want to start a new repository with a README file that will explain the contents of our repository. You can learn more about GitHub’s <a href="https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes">readme recommendations here</a>.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>touch README.md</code></pre>
</details>
<p><strong>8. Now check the status again. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>git status</code></pre>
</details>
<p>What does it look like? How is git currently treating this file?</p>
<p><strong>9. Add this file to your repository, and commit your changes. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>git add README.md
git commit -m &quot;added readme&quot; </code></pre>
</details>
<p><strong>10. Make a <code>.gitignore</code> file.</strong></p>
<p>These files are very important for bioinformatics work. Git is intended to track your small text files, you should not track your very large raw data files. Add them to the <code>.gitignore</code> to prevent this tracking.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>echo &quot;sample.fastq.gz&quot; &gt; .gitignore 
git add .gitignore 
git commit -m &quot;added gitignore&quot; </code></pre>
</details>
<p>This file is a hidden file. You won’t see it if you run a normal ls command. Instead, run <code>ls -a</code> to see the hidden files.</p>
<p><strong>11. Make a change to your README.md.</strong></p>
<p>Open the file and add some information about the folder.</p>
<p>Example:</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code># EBIO4100

This is a repository for EBIO4100.

## Contents 

This repo contains test files used to learn Git. 

## Files ignored by Git

Files ignored by Git are listed in the `.gitignore` file. 
</code></pre>
</details>
<p>These headers marked with <code>#</code> are not comments. This is a special syntax called Markdown. Markdown is a way to add formatting to simple text files that can then be interpreted by other programs. It works great for git because changes to these plain text files can be easily tracked. It can also be directly interpreted by GitHub. Here is a <a href="https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax">link for more Markdown options.</a></p>
<p><strong>12. Now, track these changes.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>git add README.md
git commit -m &quot;edited readme&quot; </code></pre>
</details>
<p><strong>13. See what has changed in this file.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>git diff --cached</code></pre>
</details>
<p>If you have a GitHub account, it can be connected to this git repository using the git config options, which are described by the module and also this <a href="https://kbroman.org/github_tutorial/pages/first_time.html">tutorial</a>.</p>
</div>
<div id="explore-github" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Explore github<a href="reproducibility.html#explore-github" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>GitHub is a platform where coders can store, share, and collaborate on code.</p>
<p>You can clone repositories from GitHub to your computer or the cluster.</p>
<p><strong>14. Try cloning a sample repository to your scratch folder.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>cd /scratch/alpine/usrname
git clone https://github.com/mfrayer/ebio4100_sample_repo.git
</code></pre>
</details>
<p>What is in your folder now?</p>
<p><strong>15. Visit the <a href="https://github.com/mfrayer/bioinformatics_in_the_mountains">GitHub repository for this book</a>.</strong></p>
<p>What information can you see about the repository? What languages was it written in? How many times have I deployed updates to the website?</p>
<p>There is also a place to note issues or make pull requests (suggested changes to the contents of the repository).</p>
<p><strong>16. Optional: If you want to make a GitHub account (or if you have one already), go ahead and try to make a pull request for a suggested change to this book! </strong></p>
</div>
<div id="assignment-7" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> Assignment<a href="reproducibility.html#assignment-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Select one of the scripts that you turned in last week and annotate it. Copy it to the <code>$turnin/day_8</code> folder. Remember to include information about where the data came from, as well as the versions for the modules you used.</p>
<div class="learning" data="latex=&quot;&quot;">
<p>Here is more information on the programs we used today:</p>
<p><a href="https://git-scm.com/about/branching-and-merging">Git</a></p>
<p><a href="https://github.com/about">GitHub</a></p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="annotation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="blast-and-genome-browser.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/09-reproducibility.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection",
    "scroll_highlight": true
  }
});
});
</script>

</body>

</html>
