<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Activity 4 Variant calling | EBIO4100: Bioinformatics in the Mountains</title>
  <meta name="description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Activity 4 Variant calling | EBIO4100: Bioinformatics in the Mountains" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/logo_hagar.JPG" />
  <meta property="og:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="github-repo" content="mfrayer/bioinformatics_in_the_mountains" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Activity 4 Variant calling | EBIO4100: Bioinformatics in the Mountains" />
  
  <meta name="twitter:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="twitter:image" content="/images/logo_hagar.JPG" />

<meta name="author" content="Megan Frayer" />


<meta name="date" content="2025-07-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="alignment-of-whole-genome-data.html"/>
<link rel="next" href="variant-filtering.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EBIO4100: Bioinformatics in the Mountains</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><i class="fa fa-check"></i>Welcome to EBIO4100: Bioinformatics in the Mountains!</a>
<ul>
<li class="chapter" data-level="0.1" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-use-this-ebook"><i class="fa fa-check"></i><b>0.1</b> How to use this ebook</a></li>
<li class="chapter" data-level="0.2" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#goalslearning-objectives"><i class="fa fa-check"></i><b>0.2</b> Goals/Learning Objectives</a></li>
<li class="chapter" data-level="0.3" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-succeed-in-this-course"><i class="fa fa-check"></i><b>0.3</b> How to succeed in this course:</a></li>
<li class="chapter" data-level="0.4" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#important-links"><i class="fa fa-check"></i><b>0.4</b> Important Links</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>1</b> Computer set up and Introduction to the command line</a>
<ul>
<li class="chapter" data-level="1.1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#get-oriented-to-the-cluster"><i class="fa fa-check"></i><b>1.1</b> Get oriented to the cluster</a></li>
<li class="chapter" data-level="1.2" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#first-command"><i class="fa fa-check"></i><b>1.2</b> First command</a></li>
<li class="chapter" data-level="1.3" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#how-to-navigate-a-file-system"><i class="fa fa-check"></i><b>1.3</b> How to navigate a file system</a></li>
<li class="chapter" data-level="1.4" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#basic-file-manipulation"><i class="fa fa-check"></i><b>1.4</b> Basic file manipulation</a></li>
<li class="chapter" data-level="1.5" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#assignment"><i class="fa fa-check"></i><b>1.5</b> Assignment</a></li>
<li class="chapter" data-level="1.6" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#if-we-have-extra-time"><i class="fa fa-check"></i><b>1.6</b> If we have extra time…</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html"><i class="fa fa-check"></i><b>2</b> FASTA files and visual alignment with MAFFT</a>
<ul>
<li class="chapter" data-level="2.1" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#exploring-a-fasta-file"><i class="fa fa-check"></i><b>2.1</b> Exploring a fasta file</a></li>
<li class="chapter" data-level="2.2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#visual-alignment"><i class="fa fa-check"></i><b>2.2</b> Visual Alignment</a></li>
<li class="chapter" data-level="2.3" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#mafft-multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.3</b> MAFFT Multiple Sequence Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#assignment-1"><i class="fa fa-check"></i><b>2.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html"><i class="fa fa-check"></i><b>3</b> Alignment of whole genome data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#environmental-variables"><i class="fa fa-check"></i><b>3.1</b> Environmental variables</a></li>
<li class="chapter" data-level="3.2" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#explore-the-sequencing-data-and-reference-genome"><i class="fa fa-check"></i><b>3.2</b> Explore the sequencing data and reference genome</a></li>
<li class="chapter" data-level="3.3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#fastqc"><i class="fa fa-check"></i><b>3.3</b> FASTQC</a></li>
<li class="chapter" data-level="3.4" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#trimming"><i class="fa fa-check"></i><b>3.4</b> Trimming</a></li>
<li class="chapter" data-level="3.5" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#alignment-with-bwa"><i class="fa fa-check"></i><b>3.5</b> Alignment with BWA</a></li>
<li class="chapter" data-level="3.6" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#read-groups"><i class="fa fa-check"></i><b>3.6</b> Read Groups</a></li>
<li class="chapter" data-level="3.7" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#how-good-was-our-alignment"><i class="fa fa-check"></i><b>3.7</b> How good was our alignment?</a></li>
<li class="chapter" data-level="3.8" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#assignment-2"><i class="fa fa-check"></i><b>3.8</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>4</b> Variant calling</a>
<ul>
<li class="chapter" data-level="4.1" data-path="variant-calling.html"><a href="variant-calling.html#remove-duplicates-from-the-bam-file"><i class="fa fa-check"></i><b>4.1</b> Remove duplicates from the BAM file</a></li>
<li class="chapter" data-level="4.2" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-gatk"><i class="fa fa-check"></i><b>4.2</b> Call Variants with GATK</a></li>
<li class="chapter" data-level="4.3" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-bcftools"><i class="fa fa-check"></i><b>4.3</b> Call variants with bcftools</a></li>
<li class="chapter" data-level="4.4" data-path="variant-calling.html"><a href="variant-calling.html#are-they-all-the-same"><i class="fa fa-check"></i><b>4.4</b> Are they all the same?</a></li>
<li class="chapter" data-level="4.5" data-path="variant-calling.html"><a href="variant-calling.html#assignment-3"><i class="fa fa-check"></i><b>4.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>5</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-filtering.html"><a href="variant-filtering.html#introduction-to-bcftools"><i class="fa fa-check"></i><b>5.1</b> Introduction to BCFtools</a></li>
<li class="chapter" data-level="5.2" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-quality"><i class="fa fa-check"></i><b>5.2</b> Filter for quality</a></li>
<li class="chapter" data-level="5.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-biallelic-snps"><i class="fa fa-check"></i><b>5.3</b> Filter for biallelic SNPs</a></li>
<li class="chapter" data-level="5.4" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-frequency"><i class="fa fa-check"></i><b>5.4</b> Filter for frequency</a></li>
<li class="chapter" data-level="5.5" data-path="variant-filtering.html"><a href="variant-filtering.html#bcftools-query"><i class="fa fa-check"></i><b>5.5</b> bcftools query</a></li>
<li class="chapter" data-level="5.6" data-path="variant-filtering.html"><a href="variant-filtering.html#assignment-4"><i class="fa fa-check"></i><b>5.6</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EBIO4100: Bioinformatics in the Mountains</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-calling" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Activity 4</span> Variant calling<a href="variant-calling.html#variant-calling" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Today we will use the bam file that we made in the last class to call variants (<code>mountain_chickadee.filtered.sorted.bam</code>). We will call variants using two popular approaches.
If your file was deleted, you can copy a new one from the shared folder.</p>
<p><strong>1. Make folders for today, and copy your file into it. Then, start an interactive job.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>cd /scratch/alpine/usrname
mkdir day_4 
cp $shared/day_4/* day_4/
cd day_4 

sinteractive --partition=alpine -t 4:00:00</code></pre>
</details>
<div id="remove-duplicates-from-the-bam-file" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Remove duplicates from the BAM file<a href="variant-calling.html#remove-duplicates-from-the-bam-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For many analyses, the last thing we need to do with the BAM file is remove duplicates. Reads sometimes get duplicated during library prep (the chemical process we use to prepare isolated DNA for sequencing). This will show up in our BAM file as reads that are completely identical from start to end. For most analyses, we want to remove these because they might give us false information about the frequency of variants they contain.</p>
<p>To remove duplicates, we will use Picard. Picard is a large and powerful program, you can look at <a href="https://broadinstitute.github.io/picard/">its manual</a> for more information. After removing duplicates, we will need to re-index the file.</p>
<p><strong>2. Remove duplicates and reindex the file.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>module load picard 
picard MarkDuplicates -INPUT mountain_chickadee.filtered.sorted.bam -OUTPUT mountain_chickadee.filtered.sorted.nodup.bam -METRICS_FILE mountain_chickadee.duplicate.metrics -REMOVE_DUPLICATES true -ASSUME_SORTED true -TMP_DIR tmp_mountain_chickadee -VALIDATION_STRINGENCY LENIENT

module load samtools 
samtools index mountain_chickadee.filtered.sorted.nodup.bam</code></pre>
</details>
<p>In the case of this small set of reads, there are no duplicates, so we have not changed anything in the file. If you look at the file <code>mountain_chickadee.duplicate.metrics</code> you can see the counts of reads that it examined.</p>
</div>
<div id="call-variants-with-gatk" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Call Variants with GATK<a href="variant-calling.html#call-variants-with-gatk" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The first variant caller we will try is called GATK. This is one of the most common variant callers. It requires two steps. First, <code>HaplotypeCaller</code> is used to create a GVCF file. Second, <code>GenotypeGVCFs</code> will make out final VCF file. If we had multiple samples, we would run each separately in <code>HaplotypeCaller</code>, and then combine them before running <code>GenotypeGVCFs</code>.</p>
<p><strong>3. Load the module and run HaplotypeCaller.</strong></p>
<p>Note: GATK provides cluster-specific instructions when you load the module.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
module load gatk

gatk --java-options &quot;-Djava.io.tmpdir=$SLURM_SCRATCH&quot; HaplotypeCaller -R /scratch/alpine/mefr3284/ebio4100/reference/moch.dovetail.reference.hap1.scaffolds.renamed.fasta -I mountain_chickadee.filtered.sorted.nodup.bam -O moch_gatk.g.vcf.gz -ERC GVCF
</code></pre>
</details>
<p><strong>4. Run GenotypeGVCFs. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
gatk --java-options &quot;-Djava.io.tmpdir=$SLURM_SCRATCH&quot; GenotypeGVCFs -R /scratch/alpine/mefr3284/ebio4100/reference/moch.dovetail.reference.hap1.scaffolds.renamed.fasta -V moch_gatk.g.vcf.gz -O moch_gatk.vcf.gz
</code></pre>
</details>
<p>Notice that an index file was also made! (<code>moch_gatk.g.vcf.gz.tbi</code>)</p>
<p>GATK has really good documentation, including this <a href="https://gatk.broadinstitute.org/hc/en-us">Best Practices workflow</a> that describes the gold standard for genome processing.</p>
</div>
<div id="call-variants-with-bcftools" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Call variants with bcftools<a href="variant-calling.html#call-variants-with-bcftools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another popular variant caller is <code>bcftools mpileup</code>. We will use the program <code>bcftools</code> a lot for processing VCF files. We are going to run both variant callers and compare the output.</p>
<p><strong>5. Run bcftools mpileup and call.</strong></p>
<p>There are also two steps to this variant calling, but we can pipe from one to the other without making the large intermediate file.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
module load bcftools

bcftools mpileup -f /scratch/alpine/mefr3284/ebio4100/reference/moch.dovetail.reference.hap1.scaffolds.renamed.fasta mountain_chickadee.filtered.sorted.nodup.bam -a DP,AD -Ou | bcftools call -mv -Oz -f GQ -o moch_bcftools.vcf.gz

tabix -fp vcf moch_bcftools.vcf.gz
</code></pre>
</details>
</div>
<div id="are-they-all-the-same" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Are they all the same?<a href="variant-calling.html#are-they-all-the-same" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>6. Look at the files side by side.</strong></p>
<p>What is similar about them, and what is different? You can check their file sizes, use <code>less</code> to look at each one, and/or open each one in the OnDemand portal and look at them side-by-side that way.</p>
<p><strong>7. Run <code>bcftools stats</code> to calculate differences. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
bcftools stats -s - moch_gatk.vcf.gz moch_bcftools.vcf.gz &gt; stats.txt
</code></pre>
</details>
<p>Scroll through the output. There is a ton of information here, divided into sections.</p>
<p><strong>8. Look at the ID section.</strong></p>
<p>One of the first sections is called “Definition of sets”. Here, ID numbers are described. The first two (0 and 1) should be describe things that are unique to each of the files, and the third one (2) should describe things that are the same between the files.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
grep -B2 &quot;^ID\s&quot; stats.txt
</code></pre>
</details>
<p>This command returns lines that start with (^) ID followed by white space. The -B2 flag returns the two lines prior to any lines that match the criteria.</p>
<p><strong>9. Next, look at the section “Summary Numbers”. How many records do the VCF files have in common?</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
grep -B1 &quot;^SN\s&quot; stats.txt
</code></pre>
</details>
<p><strong>10. Next, look at the section “Per-sample counts” or “PSC”. </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
grep PSC stats.txt
</code></pre>
</details>
<p>Below is a sample table (yours may or may not be identical). You will notice that <code>bcftools</code> called many more sites, and the category with the least agreement is the calling of indels (insertions or deletions). Calling indels is very tricky, and a big way for programs to diverge, especially when coverage is low. These files have artificially low coverage, and we will work with the higher coverage version of this dataset later. However, if this was our dataset, we would want to filter this dataset carefully.</p>
<style type="text/css">
.scrolling {
  max-width: 1000px;
  overflow-x: auto;
}
</style>
<pre style="max-width: 1000px;"><code>|PSC|[2]id|[3]sample|[4]nRefHom|[5]nNonRefHom|[6]nHets|[7]nTransitions|[8]nTransversions|[9]nIndels|[10]average depth|[11]nSingletons|[12]nHapRef|[13]nHapAlt|[14]nMissing|
|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
|PSC|0|moch21|0|94|0|43|51|227|1.9|321|0|0|0|
|PSC|1|moch21|0|5263|0|3367|1896|202|1.0|5465|0|0|0|
|PSC|2|moch21|0|2745|1|1878|868|30|2.0|2776|0|0|0|</code></pre>
<p>So which of these variant sets is right?</p>
<p>The hard truth is that we don’t know the real genotypes– both of these files provide different hypotheses for the genotypes based on the alignment we provided. It is important to think carefully about your data, the analyses you are trying to do, and the assumptions of the model in order to decide which program is right for your analysis.</p>
</div>
<div id="assignment-3" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Assignment<a href="variant-calling.html#assignment-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create a script that will remove duplicates and call variants in the BAM file you made in yesterday’s assignment (<code>20H102_S21_L002_R2_001.filtered.sorted.bam</code>). You can use either program. When you are done, copy it to the folder <code>$turnin/day_4/</code>. Make sure you add your name to the file name! Try to run it on the cluster, it may take a few hours.</p>
<p><em>Hint 1:</em></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
Your header could look like: 
  
#!/bin/bash

#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --time=6:00:00
#SBATCH --partition=amilan 
#SBATCH --output=vcfs.out
#SBATCH --qos=normal
</code></pre>
</details>
<p><em>Hint 2:</em></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
Make sure you have the names and locations of the files correct! 
</code></pre>
</details>
<div class="learning" data="latex=&quot;&quot;">
<p>If you would like more information on the programs we tried today, here are links to check out.</p>
<p><a href="https://gatk.broadinstitute.org/hc/en-us">GATK User Guide</a></p>
<p><a href="https://samtools.github.io/bcftools/bcftools.html">BCFtools Manual</a></p>
<p><a href="https://broadinstitute.github.io/picard/">Picard Documentation</a></p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="alignment-of-whole-genome-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-filtering.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-variant-calling.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection",
    "scroll_highlight": true
  }
});
});
</script>

</body>

</html>
