<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Activity 5 Variant filtering | EBIO4100: Bioinformatics in the Mountains</title>
  <meta name="description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Activity 5 Variant filtering | EBIO4100: Bioinformatics in the Mountains" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/logo_hagar.JPG" />
  <meta property="og:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="github-repo" content="mfrayer/bioinformatics_in_the_mountains" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Activity 5 Variant filtering | EBIO4100: Bioinformatics in the Mountains" />
  
  <meta name="twitter:description" content="This bookdown organizes the course materials for EBIO4100: Intro to Bioinformatics, taught Summer 2025." />
  <meta name="twitter:image" content="/images/logo_hagar.JPG" />

<meta name="author" content="Megan Frayer" />


<meta name="date" content="2025-07-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="variant-calling.html"/>
<link rel="next" href="basic-phylogenetics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EBIO4100: Bioinformatics in the Mountains</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><i class="fa fa-check"></i>Welcome to EBIO4100: Bioinformatics in the Mountains!</a>
<ul>
<li class="chapter" data-level="0.1" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-use-this-ebook"><i class="fa fa-check"></i><b>0.1</b> How to use this ebook</a></li>
<li class="chapter" data-level="0.2" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#goalslearning-objectives"><i class="fa fa-check"></i><b>0.2</b> Goals/Learning Objectives</a></li>
<li class="chapter" data-level="0.3" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#how-to-succeed-in-this-course"><i class="fa fa-check"></i><b>0.3</b> How to succeed in this course:</a></li>
<li class="chapter" data-level="0.4" data-path="welcome-to-ebio4100-bioinformatics-in-the-mountains.html"><a href="welcome-to-ebio4100-bioinformatics-in-the-mountains.html#important-links"><i class="fa fa-check"></i><b>0.4</b> Important Links</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>1</b> Computer set up and Introduction to the command line</a>
<ul>
<li class="chapter" data-level="1.1" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#get-oriented-to-the-cluster"><i class="fa fa-check"></i><b>1.1</b> Get oriented to the cluster</a></li>
<li class="chapter" data-level="1.2" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#first-command"><i class="fa fa-check"></i><b>1.2</b> First command</a></li>
<li class="chapter" data-level="1.3" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#how-to-navigate-a-file-system"><i class="fa fa-check"></i><b>1.3</b> How to navigate a file system</a></li>
<li class="chapter" data-level="1.4" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#basic-file-manipulation"><i class="fa fa-check"></i><b>1.4</b> Basic file manipulation</a></li>
<li class="chapter" data-level="1.5" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#assignment"><i class="fa fa-check"></i><b>1.5</b> Assignment</a></li>
<li class="chapter" data-level="1.6" data-path="computer-set-up-and-introduction-to-the-command-line.html"><a href="computer-set-up-and-introduction-to-the-command-line.html#if-we-have-extra-time"><i class="fa fa-check"></i><b>1.6</b> If we have extra time…</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html"><i class="fa fa-check"></i><b>2</b> FASTA files and visual alignment with MAFFT</a>
<ul>
<li class="chapter" data-level="2.1" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#exploring-a-fasta-file"><i class="fa fa-check"></i><b>2.1</b> Exploring a fasta file</a></li>
<li class="chapter" data-level="2.2" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#visual-alignment"><i class="fa fa-check"></i><b>2.2</b> Visual Alignment</a></li>
<li class="chapter" data-level="2.3" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#mafft-multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.3</b> MAFFT Multiple Sequence Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="fasta-files-and-visual-alignment-with-mafft.html"><a href="fasta-files-and-visual-alignment-with-mafft.html#assignment-1"><i class="fa fa-check"></i><b>2.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html"><i class="fa fa-check"></i><b>3</b> Alignment of whole genome data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#environmental-variables"><i class="fa fa-check"></i><b>3.1</b> Environmental variables</a></li>
<li class="chapter" data-level="3.2" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#explore-the-sequencing-data-and-reference-genome"><i class="fa fa-check"></i><b>3.2</b> Explore the sequencing data and reference genome</a></li>
<li class="chapter" data-level="3.3" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#fastqc"><i class="fa fa-check"></i><b>3.3</b> FASTQC</a></li>
<li class="chapter" data-level="3.4" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#trimming"><i class="fa fa-check"></i><b>3.4</b> Trimming</a></li>
<li class="chapter" data-level="3.5" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#alignment-with-bwa"><i class="fa fa-check"></i><b>3.5</b> Alignment with BWA</a></li>
<li class="chapter" data-level="3.6" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#read-groups"><i class="fa fa-check"></i><b>3.6</b> Read Groups</a></li>
<li class="chapter" data-level="3.7" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#how-good-was-our-alignment"><i class="fa fa-check"></i><b>3.7</b> How good was our alignment?</a></li>
<li class="chapter" data-level="3.8" data-path="alignment-of-whole-genome-data.html"><a href="alignment-of-whole-genome-data.html#assignment-2"><i class="fa fa-check"></i><b>3.8</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>4</b> Variant calling</a>
<ul>
<li class="chapter" data-level="4.1" data-path="variant-calling.html"><a href="variant-calling.html#remove-duplicates-from-the-bam-file"><i class="fa fa-check"></i><b>4.1</b> Remove duplicates from the BAM file</a></li>
<li class="chapter" data-level="4.2" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-gatk"><i class="fa fa-check"></i><b>4.2</b> Call Variants with GATK</a></li>
<li class="chapter" data-level="4.3" data-path="variant-calling.html"><a href="variant-calling.html#call-variants-with-bcftools"><i class="fa fa-check"></i><b>4.3</b> Call variants with bcftools</a></li>
<li class="chapter" data-level="4.4" data-path="variant-calling.html"><a href="variant-calling.html#are-they-all-the-same"><i class="fa fa-check"></i><b>4.4</b> Are they all the same?</a></li>
<li class="chapter" data-level="4.5" data-path="variant-calling.html"><a href="variant-calling.html#assignment-3"><i class="fa fa-check"></i><b>4.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>5</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-filtering.html"><a href="variant-filtering.html#introduction-to-bcftools"><i class="fa fa-check"></i><b>5.1</b> Introduction to BCFtools</a></li>
<li class="chapter" data-level="5.2" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-quality"><i class="fa fa-check"></i><b>5.2</b> Filter for quality</a></li>
<li class="chapter" data-level="5.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-biallelic-snps"><i class="fa fa-check"></i><b>5.3</b> Filter for biallelic SNPs</a></li>
<li class="chapter" data-level="5.4" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-for-frequency"><i class="fa fa-check"></i><b>5.4</b> Filter for frequency</a></li>
<li class="chapter" data-level="5.5" data-path="variant-filtering.html"><a href="variant-filtering.html#bcftools-query"><i class="fa fa-check"></i><b>5.5</b> bcftools query</a></li>
<li class="chapter" data-level="5.6" data-path="variant-filtering.html"><a href="variant-filtering.html#assignment-4"><i class="fa fa-check"></i><b>5.6</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html"><i class="fa fa-check"></i><b>6</b> Basic Phylogenetics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#upgma-tree"><i class="fa fa-check"></i><b>6.1</b> UPGMA Tree</a></li>
<li class="chapter" data-level="6.2" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#mafft"><i class="fa fa-check"></i><b>6.2</b> MAFFT</a></li>
<li class="chapter" data-level="6.3" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#iq-tree"><i class="fa fa-check"></i><b>6.3</b> IQ-TREE</a></li>
<li class="chapter" data-level="6.4" data-path="basic-phylogenetics.html"><a href="basic-phylogenetics.html#assignment-5"><i class="fa fa-check"></i><b>6.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="annotation.html"><a href="annotation.html"><i class="fa fa-check"></i><b>7</b> Annotation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="annotation.html"><a href="annotation.html#find-features-in-a-sequence"><i class="fa fa-check"></i><b>7.1</b> Find features in a sequence</a></li>
<li class="chapter" data-level="7.2" data-path="annotation.html"><a href="annotation.html#intro-to-bedtools"><i class="fa fa-check"></i><b>7.2</b> Intro to BEDtools</a></li>
<li class="chapter" data-level="7.3" data-path="annotation.html"><a href="annotation.html#gene-ontology"><i class="fa fa-check"></i><b>7.3</b> Gene Ontology</a></li>
<li class="chapter" data-level="7.4" data-path="annotation.html"><a href="annotation.html#assignment-6"><i class="fa fa-check"></i><b>7.4</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reproducibility.html"><a href="reproducibility.html"><i class="fa fa-check"></i><b>8</b> Reproducibility</a>
<ul>
<li class="chapter" data-level="8.1" data-path="reproducibility.html"><a href="reproducibility.html#how-do-you-draw-shapes"><i class="fa fa-check"></i><b>8.1</b> How do you draw shapes?</a></li>
<li class="chapter" data-level="8.2" data-path="reproducibility.html"><a href="reproducibility.html#practice-annotating-code"><i class="fa fa-check"></i><b>8.2</b> Practice annotating code</a></li>
<li class="chapter" data-level="8.3" data-path="reproducibility.html"><a href="reproducibility.html#version-control-software"><i class="fa fa-check"></i><b>8.3</b> Version Control Software</a></li>
<li class="chapter" data-level="8.4" data-path="reproducibility.html"><a href="reproducibility.html#explore-github"><i class="fa fa-check"></i><b>8.4</b> Explore github</a></li>
<li class="chapter" data-level="8.5" data-path="reproducibility.html"><a href="reproducibility.html#assignment-7"><i class="fa fa-check"></i><b>8.5</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EBIO4100: Bioinformatics in the Mountains</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-filtering" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Activity 5</span> Variant filtering<a href="variant-filtering.html#variant-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="introduction-to-bcftools" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Introduction to BCFtools<a href="variant-filtering.html#introduction-to-bcftools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Yesterday we used the variant caller built into BCFtools, but BCFtools is actually a very large program with a lot of functions for processing VCF files.</p>
<p>Take a minute to look at the <a href="https://samtools.github.io/bcftools/bcftools.html">manual</a> and see the different functions.</p>
<p><strong>1. Make folders for today, and copy your file into it. Then, start an interactive job.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>cd /scratch/alpine/usrname
mkdir day_5
cp $shared/day_5/* day_5/
cd day_5 

sinteractive --partition=amilan -t 4:00:00</code></pre>
</details>
</div>
<div id="filter-for-quality" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Filter for quality<a href="variant-filtering.html#filter-for-quality" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>2. Filter your VCF based on quality.</strong></p>
<p>Flags <code>-i</code> and <code>-e</code> are inverses of each other (“include” vs “exclude”), so the two commands below should be equivalent.</p>
<p>Note that on this cluster, we need to load the samtools module in order to get access to <code>bgzip</code>. BCFtools prefers bgzipped VCF files.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>module load bcftools
module load samtools 

bcftools view -i &quot;QUAL&gt;30&quot; moch_bcftools.vcf.gz | bgzip -c &gt; moch.filteredI.vcf.gz

bcftools view -e &quot;QUAL&lt;30&quot; moch_bcftools.vcf.gz | bgzip -c &gt; moch.filteredE.vcf.gz
</code></pre>
</details>
<p><strong>3. How did the file change? How many sites are in the file now? </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>bcftools view -H moch_bcftools.vcf.gz | wc -l
bcftools view -H moch.filteredI.vcf.gz | wc -l
bcftools view -H moch.filteredE.vcf.gz | wc -l</code></pre>
</details>
<p><strong>4. What if we use filter instead of view?</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>bcftools filter -i &quot;QUAL&gt;30&quot; moch_bcftools.vcf.gz | bgzip -c &gt; moch.filteredI2.vcf.gz
bcftools view -H moch_bcftools.vcf.gz | wc -l
bcftools view -H moch.filteredI.vcf.gz | wc -l
bcftools view -H moch.filteredI2.vcf.gz | wc -l</code></pre>
</details>
<p>How did the file change?</p>
<p>If you’re not sure, try <code>bcftools view -r Chr8_1:436750-436760</code> with both files.</p>
</div>
<div id="filter-for-biallelic-snps" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Filter for biallelic SNPs<a href="variant-filtering.html#filter-for-biallelic-snps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Many statistical analyses will require you to select only the biallelic SNPs (i.e. records where there are two possible variants).</p>
<p><strong>5. Filter for biallelic SNPs.</strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>bcftools view --type snps -m 2 -M 2 moch.filteredE.vcf.gz &gt; moch.filtered.bSNPs.vcf.gz </code></pre>
</details>
<p><strong>6. How did the file change? How many sites are in the file now? </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>bcftools view -H moch.filteredE.vcf.gz | wc -l
bcftools view -H moch.filtered.bSNPs.vcf.gz | wc -l</code></pre>
</details>
</div>
<div id="filter-for-frequency" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Filter for frequency<a href="variant-filtering.html#filter-for-frequency" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Many analyses will be affected by low frequency variants (i.e. variants that only occur in a small number of haplotypes in your sample). We can also use <code>bcftools</code> to filter based on frequency.</p>
<p>The commands below will not be useful on our VCF file because it only contains one individual. Instead of running them, look up the flags -ac and -q in the <a href="https://samtools.github.io/bcftools/bcftools.html">BCFtools Manual</a>.</p>
<p><strong>7. What would each of these commands do in a multisample VCF? What will they do if we run them on our file? </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>bcftools view -c 2 moch.filtered.vcf.gz &gt; moch.filtered.minAC2.vcf.gz 
bcftools view -q &gt; 0.5 moch.filtered.vcf.gz &gt; moch.filtered.minFreq05.vcf.gz </code></pre>
</details>
</div>
<div id="bcftools-query" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> bcftools query<a href="variant-filtering.html#bcftools-query" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>bcftools query</code> is a really useful tool for creating custom tables from your VCF files.</p>
<p>*8. Make a table that shows the allele depth for each variant. **</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>bcftools query -f &#39;%CHROM %POS %REF %ALT [%AD ]\n&#39; moch.filteredI.vcf.gz &gt; moch.alleleDepth.table</code></pre>
</details>
<p>*9. What is the distribution of allele depths that were used to call these variants? **</p>
<p>Once we make a custom table with <code>query</code>, we can easily process them with other tools. You might use it as inputs for customs scripts in R or python, or you can even parse them with command line tools.</p>
<p>There are three new commands that can help us answer the question posed here.
1. <code>cut</code> can be used to split columns in tab-delimited files. Since we made our table space-delimited, we specify that with the <code>-d" "</code> flag. The flag <code>-f5</code> tells cut to return only column 5.
2. <code>sort</code> sorts the rows. They can be sorted numerically, but in this case we just want the matching rows to group together.
3. <code>uniq</code> selects the unique rows from the input. <code>uniq</code> only compares one line to the next, so you often need to combine <code>uniq</code> with <code>sort</code> to make sure that matching lines are next to each other. Combining <code>sort</code> and <code>unique</code> is often useful for checking for errors in files. The <code>-c</code> flag with <code>uniq</code> will return the number of lines of each type, rather than just the lines themselves.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>cut -d&quot; &quot; -f5 moch.alleleDepth.table | sort | uniq -c  </code></pre>
</details>
<p><strong>10. How many alternative alleles are A, T, G, or C? </strong></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>cut -d&quot; &quot; -f4 moch.alleleDepth.table | grep &#39;^[ATGC]$&#39; | sort | uniq -c</code></pre>
</details>
</div>
<div id="assignment-4" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Assignment<a href="variant-filtering.html#assignment-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create a script that will filter for quality and remove indels in the VCF file you created yesterday. When you are done, copy it to the folder <code>$turnin/day_5/</code>. Make sure you add your name to the file name!</p>
<p><em>Hint 1:</em></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>
Your header could look like: 
  
#!/bin/bash

#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --time=0:30:00
#SBATCH --partition=amilan 
#SBATCH --output=filtering.out
#SBATCH --qos=normal  
</code></pre>
</details>
<p><em>Hint 2:</em></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<pre class="foldable"><code>You can combine multiple filtering flags into one bcftools command.  </code></pre>
</details>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="variant-calling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="basic-phylogenetics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/06-variant-filtering.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection",
    "scroll_highlight": true
  }
});
});
</script>

</body>

</html>
